plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"google.golang.org/grpc"
+)
+
+type OpenTelemetry struct {
+	ServiceAddress string          `toml:"service_address"`
+	Timeout        config.Duration `toml:"timeout"`
+
+	MetricsSchema string `toml:"metrics_schema"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	metricsConverter     *influx2otel.LineProtocolToOtelMetrics
+	grpcClientConn       *grpc.ClientConn
+	metricsServiceClient otlpcollectormetrics.MetricsServiceClient
+}
+
+func newOpenTelemetry() *OpenTelemetry {
+	return &OpenTelemetry{
+		ServiceAddress: "localhost:4317",
+		Timeout:        config.Duration(5 * time.Second),
+		MetricsSchema:  "prometheus-v1",
+	}
+}
+
+const sampleConfig = `
+  ## Override the OpenTelemetry gRPC service address:port
+  # service_address = "localhost:4317"
+
+  ## Override the default request timeout
+  # timeout = "5s"
+
+  ## Select a schema for metrics: prometheus-v1 or prometheus-v2
+  ## For more information about the alternatives, read the Prometheus input
+  ## plugin notes.
+  # metrics_schema = "prometheus-v1"
+`
+
+func (o *OpenTelemetry) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *OpenTelemetry) Description() string {
+	return "Send OpenTelemetry metrics over gRPC"
+}
+
+var metricsSchemata = map[string]common.MetricsSchema{
+	"prometheus-v1": common.MetricsSchemaTelegrafPrometheusV1,
+	"prometheus-v2": common.MetricsSchemaTelegrafPrometheusV2,
+}
+
+func (o *OpenTelemetry) Connect() error {
+	logger := &otelLogger{o.Log}
+	ms, found := metricsSchemata[o.MetricsSchema]
+	if !found {
+		return fmt.Errorf("schema '%s' not recognized", o.MetricsSchema)
+	}
+
+	metricsConverter, err := influx2otel.NewLineProtocolToOtelMetrics(logger, ms)
+	if err != nil {
+		return err
+	}
+
+	grpcClientConn, err := grpc.Dial(o.ServiceAddress, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+
+	metricsServiceClient := otlpcollectormetrics.NewMetricsServiceClient(grpcClientConn)
+
+	o.metricsConverter = metricsConverter
+	o.grpcClientConn = grpcClientConn
+	o.metricsServiceClient = metricsServiceClient
+
+	return nil
+}
+
+func (o *OpenTelemetry) Close() error {
+	return o.grpcClientConn.Close()
+}
+
+func (o *OpenTelemetry) Write(metrics []telegraf.Metric) error {
+	batch := o.metricsConverter.NewBatch()
+	for _, metric := range metrics {
+		var vType common.InfluxMetricValueType
+		switch metric.Type() {
+		case telegraf.Gauge:
+			vType = common.InfluxMetricValueTypeGauge
+		case telegraf.Untyped:
+			vType = common.InfluxMetricValueTypeUntyped
+		case telegraf.Counter:
+			vType = common.InfluxMetricValueTypeSum
+		case telegraf.Histogram:
+			vType = common.InfluxMetricValueTypeHistogram
+		case telegraf.Summary:
+			vType = common.InfluxMetricValueTypeSummary
+		default:
+			o.Log.Warnf("unrecognized metric type %Q", metric.Type())
+			continue
+		}
+		err := batch.AddPoint(metric.Name(), metric.Tags(), metric.Fields(), metric.Time(), vType)
```

Comment:
There's a panic in AddPoint that I think should just be an error. Telegraf really doesn't like panics. https://github.com/influxdata/influxdb-observability/blob/6ea7daf489f30bd9e7f7d3f7ef65e8074fb6a2ec/influx2otel/metrics_telegraf_prometheus_v1.go#L83

Reply:
Fixed.